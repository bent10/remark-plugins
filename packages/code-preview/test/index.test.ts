/// <reference types="vitest/globals" />

import { remark } from 'remark'
import remarkCodePreview from '../src/index.js'

it('should transform code blocks into code previews', async () => {
  const file = await remark().use(remarkCodePreview)
    .process(`import { Foo } from 'foo'

# Example

\`\`\`jsx title="Code title"
<Foo />
\`\`\`
`)

  expect(String(file)).toMatchSnapshot()
})

it('should use a custom template when provided in options', async () => {
  const template = `
<figure class='foo'>
<figcaption>{title}</figcaption>
<div class='foo-preview'>
{preview}
</div>
<div class='foo-code'>
{code}
</div>
</figure>
`

  const file = await remark().use(remarkCodePreview, {
    template,
    ignoreMissing: true
  }).process(`\`\`\`jsx
<Foo />
\`\`\``)

  expect(String(file)).toMatchSnapshot()
})

it('should not transform code blocks that do not match the test function', async () => {
  const inputMarkdown = `
\`\`\`python
# This is a Python code block
print('Hello, World!')
\`\`\`
    `

  const file = await remark().use(remarkCodePreview).process(inputMarkdown)

  expect(String(file)).toMatchSnapshot()
})
