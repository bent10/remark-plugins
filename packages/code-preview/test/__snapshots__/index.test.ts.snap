// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should not transform code blocks that do not match the test function 1`] = `
"\`\`\`python
# This is a Python code block
print('Hello, World!')
\`\`\`
"
`;

exports[`should transform code blocks into code previews 1`] = `
"import { Foo } from 'foo'

# Example

<figure class='preview'>
<figcaption>Code title</figcaption>
<div class='preview-showcase'>
<Foo />
</div>
<div class='preview-code'>

\`\`\`jsx title=\\"Code title\\"
<Foo />
\`\`\`

</div>
</figure>
"
`;

exports[`should use a custom template when provided in options 1`] = `
"<figure class='foo'>
<figcaption>{title}</figcaption>
<div class='foo-preview'>
<Foo />
</div>
<div class='foo-code'>

\`\`\`jsx
<Foo />
\`\`\`

</div>
</figure>
"
`;
